dashboard-api/internal/types/studyconf.go

```diff
@@ -46,28 +49,50 @@ func (sc *StudyConf) TransformForDatabase() (res []DatabaseStudyConf, err error)
 		// service use to identify the config
 		fieldName := v.Type().Field(i).Tag.Get("json")
 		// if a field is not set we dont need to add it to our array
-		// as it will just have empty data
-		if field.IsZero() {
+		// as it will just have empty data as well as if it is not one of
+		// the confs which should all be pointers
+		if field.IsZero() || field.Kind() != reflect.Ptr {
 			continue
 		}
 
-		switch field.Kind() {
-		case reflect.Struct, reflect.Slice:
-			// we marshal the full config to json to be stored
-			conf, err := json.Marshal(field.Interface())
-			if err != nil {
-				return nil, err
-			}
-			res = append(res, DatabaseStudyConf{
-				StudyID:  studyID,
-				ConfType: fieldName,
-				Conf:     conf,
-			})
+		// we marshal the full config to json to be stored
+		conf, err := json.Marshal(field.Interface())
+		if err != nil {
+			return nil, err
 		}
+		res = append(res, DatabaseStudyConf{
+			StudyID:  studyID,
+			ConfType: fieldName,
+			Conf:     conf,
+		})
 	}
 	return res, nil
 }
 
+// TransformFromDatabase gets a list of DatabaseStudyConf and iterates
+// through it, adding the relevant configs to the StudyConf struct
+func (sc *StudyConf) TransformFromDatabase(dcs []*DatabaseStudyConf) error {
+	for _, d := range dcs {
+		//TODO: Im sure we can make this more future proof by using
+		//reflect and checking the json tags
+		switch d.ConfType {
+		case "general":
```

Comment:
I would make a map just to make it slightly cleaner, as we'll keep adding, and the logic will be identical.